flexible/pubsub/src/main/java/com/example/flexible/pubsub/PubSubService.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.flexible.pubsub;
+
+import com.google.api.gax.grpc.ApiException;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.pubsub.spi.v1.Publisher;
+import com.google.cloud.pubsub.spi.v1.SubscriptionAdminClient;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.*;
+import io.grpc.Status;
+import java.util.logging.Logger;
+
+public class PubSubService {
+
+  private Publisher publisher;
+  private Logger logger = Logger.getLogger(this.getClass().getName());
+
+  /** Create a lazy initialized singleton of PubSubService */
+  private static class LazyInit {
+    private static final PubSubService INSTANCE;
+
+    static {
```

Comment:
After reading into this Lazy initialization singleton pattern. I wonder if there's a more readable solution? This wasn't clear during my first pass.